### ISSUE TITLE
node queries report wrong node/instance count (confd)

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#492]
Originally reported of Google Code with ID 436.

```
# gnt-node list
Node                 DTotal DFree MTotal MNode MFree Pinst Sinst
x.mobile.tixteam.net 119.0G    0M   7.7G  1.4G  6.3G     1     0
root@x:/var/local/log/ganeti# gnt-group list
Group Nodes Instances AllocPolicy NDParams
mine      0         0 preferred   spindle_count=1, exclusive_storage=False, oob_program=

Of course this is not possible. There cannot be 0 nodes in a group. And in this case there is an instance as well, which is not reported.

Thanks,

Guido

```
Originally added on 2013-04-24 06:45:12 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2013-04-29 10:47:14 +0000 UTC.

Added Labels: Priority-Medium
Added to Milestone: Release2.7

**Comment by Ganeti-Issues-Migrator** [comment]
```
Will look into this as well
```
Originally added on 2013-04-29 10:48:25 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2013-04-29 10:51:04 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
commit c42fbe285ffd56d70663144d6fe3f47c0b54f7bd
Author: Iustin Pop <iustin@google.com>
Date:   Mon Feb 25 17:18:36 2013 +0100

    Fix bug in group queries related to node/instance fields
    
    Since we use the primitive string type for group UUIDs, the group
    fields have a bug where we pass the group name as filter for node
    tests, whereas the nodes themselves use the group UUID. This results
    in zero node count, empty node list, and no instances being reported
    as assigned to groups.
    
    The patch fixes this and adds a test for the node count. It does some
    test generation improvement, and also cleans up whitespace issues in
    Test/G/Q/Query.hs (the functions case_queryNode_allfields,
    prop_queryGroup_noUnknown and case_queryGroup_allfields are unchanged
    but simply have indentation fixed).
    
    Signed-off-by: Iustin Pop <iustin@google.com>
    Reviewed-by: Guido Trotter <ultrotter@google.com>
    
    Cherry-pick of e7124835, fixes issue 436
    Signed-off-by: Klaus Aehlig <aehlig@google.com>
    Reviewed-by: Guido Trotter <ultrotter@google.com>
    
    Conflicts:
    	test/hs/Test/Ganeti/Objects.hs
    	test/hs/Test/Ganeti/Query/Query.hs

```
Originally added on 2013-04-30 07:58:20 +0000 UTC.

Mentioned Issue with GoogleCode ID 436 has been migrated to GitHub with ID #492.
Changed State: Fixed

**Comment by Ganeti-Issues-Migrator** [comment]
```
Marking 2.7 bugs fixed, as 2.7.0~rc1 was released, and should be ready to be tested in production environments.
```
Originally added on 2013-05-09 05:42:49 +0000 UTC.

Changed State: Released
